packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js

```diff
@@ -94,14 +91,22 @@ export default class QuanticGeneratedAnswer extends LightningElement {
     );
   };
 
+  readStoredData() {
+    try {
+      return JSON.parse(sessionStorage?.getItem(GENERATED_ANSWER_DATA_KEY));
+    } catch {
+      return {};
+    }
+  }
+
   buildHeadlessGeneratedAnswerController(engine) {
-    const storedGeneratedAnswerVisibility = sessionStorage?.getItem(
-      GENERATED_ANSWER_VISIBILITY_KEY
-    );
-    if (['true', 'false'].includes(storedGeneratedAnswerVisibility)) {
+    const storedGeneratedAnswerData = this.readStoredData();
+    const storedGeneratedAnswerVisibility =
+      storedGeneratedAnswerData?.isVisible;
+    if (typeof storedGeneratedAnswerVisibility == 'boolean') {
       return this.headless.buildGeneratedAnswer(engine, {
         initialState: {
-          isVisible: storedGeneratedAnswerVisibility === 'true' ? true : false,
+          isVisible: storedGeneratedAnswerVisibility,
```

Comment:
Good idea more simple, will update 👌
Tho in my other it would still build a generated answer if something weird is in the Session Storage, basically the code was following this logic: if there is a weird value in the local storage then create the controller without an initial state. When creating the controller that way Headless will automatically use the default value which is to by default show the generated answer.

Reply:
ah ok i think this was my view because I was checking only what was modified in last commit, I didn't see the rest of the if